### PR TITLE
add create and list registry methods

### PR DESCRIPTION
**Pull request by elenz97**

### CODE REVIEW

File: registry.go [hunk 1]
```diff
@@ -11,6 +11,25 @@ type RegistryClient struct {
...
+		End()
+	return CheckResponse(errs, resp, 201)
+}
+
+// ListRegistries
+// Get a list of registries
+func (s *RegistryClient) ListRegistries(name string) ([]Registry, error) {
+	var r []Registry
+	resp, _, errs := s.NewRequest(gorequest.GET, "").
+		Query(name).
```
Comment (martin-helmich) [inline registry.go:28]: What does the `name` parameter actually do? If used in a query string, shouldn't this be `Query(map[string]string{"name": name})` or something similar?

Comment (elenz97) [reply registry.go]: Yeah, it does query the API, though the suggested way is more elegant.
Found some other places where this was not happening, too.

File: user.go [hunk 2]
```diff
@@ -15,7 +16,7 @@ type UserClient struct {
 func (s *UserClient) SearchUser(usr UserMember) (UserSearchResults, error) {
 	var u UserSearchResults
 	resp, _, errs := s.NewRequest(gorequest.GET, "/search").
-		Query(fmt.Sprintf("username=%s", usr.Username)).
+		Query(map[string]string{"username": url.PathEscape(usr.Username)}).
```
Comment (martin-helmich) [inline user.go:19]: See above

File: registry.go [hunk 3]
```diff
@@ -11,12 +11,30 @@ type RegistryClient struct {
...
+		End()
+	return CheckResponse(errs, resp, 201)
+}
+
+// ListRegistries
+// Get a list of registries
+func (s *RegistryClient) ListRegistries(name string) ([]Registry, error) {
+	var r []Registry
+	resp, _, errs := s.NewRequest(gorequest.GET, "").
+		Query(map[string]string{"name": name}).
```
Comment (martin-helmich) [inline registry.go:28]: What does the API respond with, if we just pass `&name=` as query? Maybe we should consider `name == ""` as a special case in which no query should be added at all (to list _all_ registries)?

File: project.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ func (s *ProjectClient) ListProjects(opt ListProjectsOptions) ([]Project, error)
 // Check if the project name provided already exist
 func (s *ProjectClient) CheckProject(projectName string) error {
 	resp, _, errs := s.NewRequest(gorequest.HEAD, "").
-		Query(fmt.Sprintf("project_name=%s", projectName)).
+		Query(map[string]string{"project_name": url.PathEscape(projectName)}).
```
Comment (martin-helmich) [inline project.go:30]: I would actually expect `gorequest` to escape query parameters on its own in this case; maybe check, just to make sure that there is no double-escaping.

In any way, the escape function to use in this case would be `url.QueryEscape`, not `url.PathEscape` ;)

File: replication.go [hunk 5]
```diff
@@ -18,12 +19,12 @@ func (s *ReplicationClient) ListReplicationAdapters() ([]string, error) {
...
 
-// GetReplicationPolicies
+// ListReplicationPolicies
 // Get an array of matching replication policies
-func (s *ReplicationClient) GetReplicationPolicies(name string) ([]ReplicationPolicy, error) {
+func (s *ReplicationClient) ListReplicationPolicies(name string) ([]ReplicationPolicy, error) {
 	var rp []ReplicationPolicy
 	resp, _, errs := s.NewRequest(gorequest.GET, "/policies").
-		Query(name).
+		Query(map[string]string{"name": url.PathEscape(name)}).
```
Comment (martin-helmich) [inline replication.go:27]: See above.

File: registry.go (hunk 3 above)
Comment (elenz97) [reply registry.go:28]: Good point, the API seems to not require a query string on this endpoint.
When leaving it out entirely / querying any unsupported key, the response stays the same (an array of all registries):

```
/api/registries
/api/registries?foo
/api/registries?name=
```

Other endpoints show this behaviour as well, e.g. `/api/replication/policies` acts the same.

IMHO, i think we don't need to catch this here, but we should let the user know about this quirk through documentation. Your thoughts?

Comment (martin-helmich) [reply registry.go:28]: Probably doesn't make that much of a difference. 🤷 I think the general behaviour (leave `name` empty, get all registries) is valid; we _might_ go the extra step of removing the query parameter entirely if the user supplies an empty `name` query, in case the server ever decides to change its default behaviour (server-side, they're probably doing the same as us: decoding into a Go `string` and making no difference between "set, but empty" and "not set at all").

TL;DR: It's fine by me to leave it as it is.

File: project.go (hunk 4 above)
Comment (elenz97) [reply project.go:30]: Doesn't seem like gorequest does it, although i might be wrong. As we're querying a map `gorequest` passes it's content down to the following:
https://github.com/parnurzeal/gorequest/blob/a578a48e8d6ca8b01a3b18314c43c6716bb5f5a3/gorequest.go#L386

Comment (martin-helmich) [reply project.go]: I think it's alright:

https://github.com/parnurzeal/gorequest/blob/a578a48e8d6ca8b01a3b18314c43c6716bb5f5a3/gorequest.go#L1205-L1211

Basically, it passes all values into a `url.Values`, which does URL-encoding on its own. So no need to double-escape.
